src/wp-includes/http.php

```diff
@@ -589,7 +589,23 @@ function wp_http_validate_url( $url ) {
 	$host        = trim( $parsed_url['host'], '.' );
 
 	if ( ! $same_host ) {
-		if ( preg_match( '#^(([1-9]?\d|1\d\d|25[0-5]|2[0-4]\d)\.){3}([1-9]?\d|1\d\d|25[0-5]|2[0-4]\d)$#', $host ) ) {
+		$is_ipv4 = (bool) preg_match(
```

Comment:
Should this use the filter extension to check for a valid IP and then fall back to regex? Maybe that's not worth it, since it would be difficult to ever check when the filter extension is not enabled.